backend/check_stac_metadata/utils.py

```diff
@@ -117,19 +117,27 @@ def validate(self, url: str) -> None:  # pylint: disable=too-complex
                     ValidationResult.FAILED,
                     details={"message": error_message},
                 )
-                raise AssertionError(error_message)
 
             asset_dict = {"url": asset_url, "multihash": asset["checksum:multihash"]}
             self.logger.debug(dumps({"asset": asset_dict}))
             self.dataset_assets.append(asset_dict)
 
-        for link_object in url_json["links"]:
+        for link_object in object_json["links"]:
             next_url = link_object["href"]
             if next_url not in self.traversed_urls:
                 next_url_prefix = get_url_before_filename(next_url)
-                assert (
-                    url_prefix == next_url_prefix
-                ), f"“{url}” links to metadata file in different directory: “{next_url}”"
+
+                if self.traversed_urls[0] != next_url_prefix:
+                    error_message = (
+                        f"“{next_url}” exists in a different directory"
```

Comment:
As discussed, this and the asset URL error message should be the same. The code should probably also be pulled out since it can be made identical for assets and metadata URLs.